### ISSUE TITLE
strict: safe assumes incorrect CommonJS module

### ISSUE THREAD

**Issue by shellscape** [eslint/eslint#7895]
**Tell us about your environment**

* **ESLint Version:** latest
* **Node Version:** 7.3.0
* **npm Version:** 4.1.1

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

```
{
  "env": {
    "es6": true,
    "mocha": true,
    "node": true
  },
  "rules": {
    "brace-style": ["error", "stroustrup", { "allowSingleLine": false }],
    "consistent-return": ["off"],
    "comma-spacing": ["error", { "before": false, "after": true }],
    "complexity": ["off"],
    "indent": ["error", 2, { "SwitchCase": 1, "VariableDeclarator": 1 }],
    "keyword-spacing": ["error", { "after": true }],
    "max-len": ["warn", 1000],
    "no-bitwise": ["off"],
    "no-console": ["off"],
    "no-extra-parens": ["off"],
    "one-var": ["error", {
      "var": "always",
      "let": "always",
      "const": "never"
    }],
    "semi": ["error", "always"],
    "space-before-function-paren": ["error", "always"],
    "space-infix-ops": ["error", { "int32Hint": false }],
    "space-in-parens": ["error", "never"],
    "space-unary-ops": ["off"],
    "strict": ["error", "safe"]
  },
  "parserOptions": {
    "ecmaVersion": 8
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
wrapper.require([], function () {
  'use strict';
});
```

**What did you expect to happen?**

I expected ESLint to allow that strict declaration, as this is not a CommonJS module.

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint says this needs to be a global declaration. It doesn't look anything like a CommonJS module so it's confusing as to why it's insisting upon a global declaration. According to http://eslint.org/docs/rules/strict, that code should pass.

```bash
/.../test.js
   2:2  error  Use the global form of 'use strict'  strict
```

**Comment by platinumazure** [comment]
Using the `node` environment implies `globalReturn: true`. If your file isn't a CommonJS file, you shouldn't use the `node` environment in your configuration.

If you can share more about your project or its directory structure, we may be able to help you find a good way to configure ESLint for your project such that you use the `node` environment on actual CommonJS files only.

**Comment by shellscape** [comment]
Thanks for the quick reply. We're using ESLint via Gulp in a repo that has modules for use by Gulp as well as a nested directory of npm modules that are intended to be used on the front-end of a website - those modules contain code that makes use of RequireJS (via an in-house wrapper). Here's what that looks like:

```js
/
- src                  // where the requirejs modules live
   - some_module
      - js             // requirejs javascript files
      - package.json   // package for some_module
- tooling              //~20 node modules for gulp
- gulpfile.js
- package.json         // package for the repo
```

Our ESLint config is being loaded via a required module. So what we've got is a mixed environment. I completely missed that I had `"node"` specified as an `env`, that's on me. I removed that property from our config, which results in the RequireJS modules passing with the function-level strict declaration, but the node modules in `tooling` failing with a message of `Use the function form of 'use strict'`.

For the moment we've put `/* eslint strict: ["error", "global"] */` atop every node module file we have in the project, but that's not exactly a wonderful solution for maintenance as the number of files grow.

Would love to know if there are any other suggestions.

**Comment by platinumazure** [comment]
You should be able to create a nested `.eslintrc` file inside `src/some_module/js`, and in there, set the `node` environment to false. Then you can leave `"node": true` in your top-level `.eslintrc` file. I think that will work how you need it to.

This is assuming you're okay with CommonJS/Node files requiring "use strict" directives globally and your RequireJS files to require "use strict" directives in functions.

**Comment by shellscape** [comment]
Thanks much! Worked a charm. I still wonder if ESLint could be "smarter" about what's what, but with the myriad of ways to define and structure a module, that's probably an impossible ask.
